pkg/util/logging/log_location_hook.go

```diff
@@ -19,23 +19,39 @@ package logging
 import (
 	"fmt"
 	"runtime"
+	"strconv"
 	"strings"
 
 	"github.com/sirupsen/logrus"
 )
 
-const logLocationField = "logSource"
+const logSourceField = "logSource"
+const logSourcePriorityField = "@logSourcePriority"
 
 // LogLocationHook is a logrus hook that attaches location information
 // to log entries, i.e. the file and line number of the logrus log call.
-type LogLocationHook struct {
+// This hook instance will overwrite an existing entry in the `logSource`
+// field if its priority value is greater than or equal to the priority
+// value stored in the entry. The default priority value is 0. If
+// removePriorityField is true, the log priority value will be removed
+// from the entry.
+type logLocationHook struct {
+	priority            int
+	removePriorityField bool
 }
 
-func (h *LogLocationHook) Levels() []logrus.Level {
+func NewLogLocationHook(priority int, removePriorityField bool) *logLocationHook {
```

Comment:
Moving forward, let's try to avoid args where it's difficult to discern what they mean (`0, true`, `1, false`). Instead, we can do `NewLogLocationHook().WithPriority(1)` and let it to default to 0 otherwise.

Do we need the concept of removing the priority field? Do you ever envision there being more than "this log came from the plugin" and "this log came from ark"? Could we just do that with a simple marker field that the plugin logger/hook sets, and the ark logger sees it and removes it?

Reply:
I originally had something simpler like that, then thought I'd generalize to what I have now. Somehow I feel like I always end up on the other side of these kinds of decisions as you :) I suppose if someone implements a plugin framework for their plugin, my current impl might be useful..